src/vm/eventtrace.cpp

```diff
@@ -4105,21 +4103,17 @@ DWORD ETW::EnumerationLog::GetEnumerationOptionsFromRuntimeKeywords()
     {
         enumerationOptions |= ETW::EnumerationLog::EnumerationStructs::DomainAssemblyModuleUnload;
     }
-    if(ETW_TRACING_CATEGORY_ENABLED(MICROSOFT_WINDOWS_DOTNETRUNTIME_PROVIDER_Context, 
-        TRACE_LEVEL_INFORMATION, 
-        CLR_JIT_KEYWORD) &&
-        ETW_TRACING_CATEGORY_ENABLED(MICROSOFT_WINDOWS_DOTNETRUNTIME_PROVIDER_Context, 
-        TRACE_LEVEL_INFORMATION, 
-        CLR_ENDENUMERATION_KEYWORD))
```

Comment:
I don't think that CLR_STARTENUMERATION_KEYWORD and CLR_ENDENUMERATION_KEYWORD can be replaced with checks on existing events.  Both of these keywords have no events associated with them - they are used to enable and disable functionality via ETW.  What happens is that when an ETW session is modified the runtime gets a callback.  That callback can then do things.  In this case, it checks to see if it should do symbolic rundown, which makes sure that all of the appropriate data structures are dumped to the event stream so that analysis tools can do things like resolve symbols in JIT-compiled code.

I think that the right thing to do in this case is to leave these as is.  Any logging system that wants to support this capability needs to support a callback mechanism on config change, and today LTTng doesn't have that (though a prototype has been created).

When this is the last issue remaining to remove the COMPlus_EnableEventLog, we'll unconditionally define ETW_TRACING_CATEGORY_ENABLED as FALSE everywhere but on Windows, and probably do some preprocessor magic to fail builds if people try to use it.  Then, if we have a platform that can support this functionality, then we'll enable it on that platform.

Reply:
> I don't think that CLR_STARTENUMERATION_KEYWORD and CLR_ENDENUMERATION_KEYWORD can be replaced with checks on existing events.

I replaces them with the events that are generated when they are enabled.
I don't understand what/how this breaks something.